monai/apps/utils.py

```diff
@@ -327,7 +327,17 @@ def download_and_extract(
             be False.
         progress: whether to display progress bar.
     """
+    urlFilenameExtension = ''.join(Path(".", _basename(url)).resolve().suffixes)
```

Comment:
Please also stick to Pythonic variable name conventions, that is all lower case with words separated by _ typically. This variable could be instead `url_filename_ext` to align with this convention.